include/openthread/platform/radio.h

```diff
@@ -389,6 +389,15 @@ otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance);
 otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame);
 
 /**
+ * The radio driver calls this method to notify OpenThread that the transmission has started.
```

Comment:
I think its worth pointing out here in the comment section, that the thread of execution which calls this API must be the same thread that executes all of the other OT code.  I could see where a driver implementer may make the mistake of calling this as part of an ISR or other task if another task is responsible for executing the driver code.